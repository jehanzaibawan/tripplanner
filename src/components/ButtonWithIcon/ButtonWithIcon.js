import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Icon from "@material-ui/core/Icon";
import PropTypes from "prop-types";

const useStyles = makeStyles(theme => ({
  button: {
    marginRight: theme.spacing(1)
  },
  leftIcon: {
    marginRight: theme.spacing(1)
  }
}));

const ButtonWithIcon = props => {
  const classes = useStyles();

  return (
    <Button
      disabled={props.disabled}
      variant="contained"
      color={props.color}
      className={classes.button}
      onClick={props.onClick}
    >
      <Icon className={classes.leftIcon}>{props.icon}</Icon>
      {props.label}
    </Button>
  );
};

ButtonWithIcon.propTypes = {
  disabled: PropTypes.bool,
  color: PropTypes.string,
  label: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

ButtonWithIcon.defaultProps = {
  disabled: false,
  color: "default",
  onClick: () => {}
};

export default ButtonWithIcon;
