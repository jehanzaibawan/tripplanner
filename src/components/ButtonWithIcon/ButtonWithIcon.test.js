import React from "react";
import renderer from "react-test-renderer";
import ButtonWithIcon from "./ButtonWithIcon";

it("renders a snapshot", () => {
  const props = {
    label: "Sample",
    icon: "dummy"
  };
  const tree = renderer.create(<ButtonWithIcon {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
