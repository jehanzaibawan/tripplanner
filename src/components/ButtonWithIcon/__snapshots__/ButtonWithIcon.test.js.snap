// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-button-1 MuiButton-contained"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-leftIcon-2"
    >
      dummy
    </span>
    Sample
  </span>
</button>
`;
